File: app/src/main/java/com/example/daisyling/common/util/download/FileUtil.java
package com.example.daisyling.common.util.download;

import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStream;

/**
 * Created by Emily on 10/11/21
 */
public final class FileUtil {
    private FileUtil() {
    }

    public static File writeFile(String filePath, InputStream input) {
        if (input == null)
            return null;
        File file = new File(filePath);
        try (FileOutputStream fos = new FileOutputStream(file);
             InputStream ins = input) {
            byte[] b = new byte[1024];
            int len;
            while ((len = ins.read(b)) != -1) {
                fos.write(b, 0, len);
            }
            fos.flush();
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        } catch (IOException e) {
            e.printStackTrace();
        }
        return file;
    }
}

File: app/src/main/java/com/example/daisyling/common/util/download/FileUtil.kt
package com.example.daisyling.common.util.download

import java.io.*

/**
 * Created by Emily on 10/11/21
 */
object FileUtil {
    @JvmStatic
    fun writeFile(filePath: String?, input: InputStream?): File? {
        if (input == null) return null
        val file = File(filePath)
        try {
            FileOutputStream(file).use { fos ->
                input.use { ins ->
                    val b = ByteArray(1024)
                    var len: Int
                    while (ins.read(b).also { len = it } != -1) {
                        fos.write(b, 0, len)
                    }
                    fos.flush()
                }
            }
        } catch (e: FileNotFoundException) {
            e.printStackTrace()
        } catch (e: IOException) {
            e.printStackTrace()
        }
        return file
    }
}